src/ctap2/mod.rs

```diff
@@ -818,6 +820,7 @@ pub(crate) fn set_or_change_pin_helper<T: From<()>>(
 
 pub(crate) fn bio_enrollment(
     dev: &mut Device,
+    puat_result: Option<PinUvAuthResult>,
```

Comment:
Right, the caller doesn't, but the statemachine-code potentially does (esp. Error-handling).

Also, e.g. in case of `AuthenticatorConfig`, there sometimes is no Uv required, because AuthConfig often only works on a 'clean' (unprotected) tokens. With `PinUvAuthResult`, we can actually check this appropriately and skip all the puat-determination-thing in subsequent calls. With `Option<PinUvAuthToken>`, we can't differentiate between "First call, user doesn't have a PUAT yet" and "We did all the work already and know that no PUAT is required".

Last but not least, it felt a bit hacky to hand in the `PinUvAuthToken` and then place it more or less randomly in one of the possible `PinUvAuthResult`-variants to be able to reuse all the existing code.

Reply:
Ok!